template.go

```diff
@@ -99,6 +109,40 @@ func (e *Engine) Layout(key string) IEngineCore {
 // use it when you're in development and you don't want to restart
 // the application when you edit a template file.
 func (e *Engine) Reload(enabled bool) IEngineCore {
-	e.ShouldReload = enabled
+	e.Mutex.Lock()
+	defer e.Mutex.Unlock()
+	e.shouldReload = enabled
 	return e
 }
+
+// ShouldReload returns true if the templates should be reloaded
+func (e *Engine) ShouldReload() bool {
+	e.Mutex.RLock()
+	defer e.Mutex.RUnlock()
+	return e.shouldReload
+}
+
+// Loaded returns true if templates are loaded
+func (e *Engine) Loaded() bool {
+	e.Mutex.RLock()
+	defer e.Mutex.RUnlock()
+	return e.loaded
+}
+
+// SetLoaded sets the loaded status
+func (e *Engine) SetLoaded(enabled bool) IEngineCore {
```

Comment:
Was there a particular reason SetLoaded is the only operation with a non-locking option?

Reply:
`SetLoaded` is only called from `Load()` which already acquires a Lock for the full scope of the call.